clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4897,6 +4897,7 @@ bool SYCLIntegrationFooter::emit(raw_ostream &OS) {
   for (const VarDecl *VD : SpecConstants) {
     VD = VD->getCanonicalDecl();
     std::string TopShim = EmitSpecIdShims(OS, ShimCounter, VD);
+    OS << "#include <CL/sycl/detail/defines_elementary.hpp>\n";
```

Comment:
This should be done outside of loop over specialization constants

Reply:
Yes, this also needs to be added to the int-footer tests.

Reply:
Done